includes/data-stores/class-wc-data-store-wp.php

```diff
@@ -229,11 +229,19 @@ protected function get_wp_query_args( $query_vars ) {
 
 			// Build meta queries out of vars that are stored in internal meta keys.
 			if ( in_array( '_' . $key, $this->internal_meta_keys, true ) ) {
-				$wp_query_args['meta_query'][] = array(
-					'key'     => '_' . $key,
-					'value'   => $value,
-					'compare' => is_array( $value ) ? 'IN' : '=',
-				);
+				// Check for existing values if wildcard is used.
+				if ( '*' === $value ) {
+					$wp_query_args['meta_query'][] = array(
+						'key'     => '_' . $key,
+						'compare' => 'EXISTS',
```

Comment:
Just checking for 'EXISTS' doesn't quite work for what you are trying to do. For example, if you don't enter a sku for a  product it will have a value of '' (empty string) for the `_sku` meta when saved, and this query will still fetch it.

`!=` as the compare and `''` as the value may work, but I'm not sure if it will fetch products that don't have the `_sku` meta at all. In that case you'd have to use the `REGEXP` compare.